sbom/examples/container_image/from_catalog.py

```diff
@@ -0,0 +1,232 @@
+
+
+def get_image_data(image_nvr):
+    response = requests.get(nvr_api + image_nvr)
+    response.raise_for_status()
+    # This is a paged response, but we're assuming there are not 100+ images for a single
+    # container image NVR.
+    return sorted(response.json()["data"], key=lambda image: image["_id"])
+
+
+def get_rpms(image_id):
+    response = requests.get(rpm_manifest_api.format(catalog_image_id=image_id))
+    response.raise_for_status()
+    return sorted(response.json()["rpms"], key=lambda rpm: rpm["nvra"])
+
+
+def generate_sbom_for_image(image_nvr):
+    # Split to e.g. "ubi9-micro-container" and "9.4-6.1716471860"
+    image_nvr_name, *image_nvr_version = image_nvr.rsplit("-", maxsplit=2)
+    image_nvr_version = "-".join(image_nvr_version)
+
+    packages = []
+    relationships = []
+    image_index_pkg_created = False
+
+    for image in get_image_data(image_nvr):
+        catalog_image_id = image["_id"]
+        image_digest = image["image_id"]
+        content_sets = image["content_sets"]
+
+        # A container image may be available through more than one repo; collect all repos,
+        # registries they are available from, and the most specific tag for each repo image.
+        repos = set()
+        image_index_digest = ""
+        for repo in image["repositories"]:
+            repo_url = f"{repo['registry']}/{repo['repository']}"
+            tags = list(
+                sorted(
+                    [t for t in repo["tags"] if t["name"] != "latest"],
+                    # Sort by the length of the tag, ignoring "latest"; this is a very dumb
+                    # heuristic to find the most specific tag for a particular image. From tags
+                    # such as "9.4", "latest", and "9.4-6.1716471860", it will select the last one.
+                    key=lambda x: len(x["name"]),
+                    reverse=True,
+                )
+            )
+            if not tags:
+                print(f"ERROR: no usable tag found for image ID: {catalog_image_id}")
+                sys.exit(1)
+            repo_name = repo["repository"].split("/")[-1]
+            repos.add((repo_name, repo_url, tags[0]["name"]))
+            image_index_digest = repo["manifest_list_digest"].lstrip("sha256:")
+
+        if not repos or not image_index_digest:
+            print("ERROR: No repos or image index digest found for image ID: {catalog_image_id}")
+            sys.exit(1)
+
+        # Get license information from labels if it is set
+        # TODO: is this a license statement in SPDX licence format? Does it apply to the image
+        #  itself rather than its content (which is individually-licensed components?)
+        image_license = "NOASSERTION"
+        for label in image["parsed_data"]["labels"]:
+            if label["name"].lower() == "license":
+                image_license = label["value"]
+
+        # TODO: split image index into its own SBOM and produce per-arch image SBOM
```

Comment:
I'll rework the logic below to produce one SBOM file for each architecture with the root node being the arch-specific image, and one SBOM file that describes the image list and its descendant arch-specific images.

Reply:
@twaugh Alright, this is done. We now have one SBOM that represents the image index, and four separate SBOMs, one for each arch. Can you please give this one last look before I merge it?

In a separate PR I can create a separate example that highlights referred vs embedded content.

Reply:
Going to merge this as is. We can further refine the container examples in separate PRs if needed.